girder-dandi-archive/girder_dandi_archive/rest.py

```diff
@@ -95,81 +94,34 @@ def get_dandiset_owners(self, identifier, params):
 
     @access.user
     @autoDescribeRoute(
-        Description("Add Dandiset Owners")
+        Description("Set Dandiset Owners")
         .param("identifier", "Dandiset Identifier", paramType="path")
         .jsonParam(
             "owners",
-            "A JSON list of girder users to add as owners.",
+            "A JSON list of girder users to set as the owners.",
             paramType="body",
             requireArray=True,
         )
     )
     @dandiset_identifier
-    def add_dandiset_owners(self, identifier, owners, params):
+    def set_dandiset_owners(self, identifier, owners, params):
```

Comment:
As is you can easily delete yourself as an owner, or delete all owners simultaneously. This isn't catastrophic as it can be fixed by a girder admin, but it would be very obnoxious. I would recommend ensuring that this method always contains the user that dispatched the request, and also remove the `X` from the dialog for your own user.

Reply:
I agree, me and @mgrauer actually had a discussion about this exact thing. This is something I still need to add.

As far as removing the `X` from the dialog, my only hesitance there is there might be a use case where someone wants to make somebody else an owner, and remove themselves at the same time. This wouldn't be an issue as long as the list contains someone else as an owner. However this might not be something to worry about.